pkg/v1/remote/transport/bearer.go

```diff
@@ -60,8 +60,27 @@ func (bt *bearerTransport) RoundTrip(in *http.Request) (*http.Response, error) {
 	}
 	in.Header.Set("User-Agent", transportName)
 
-	// TODO(mattmoor): On 401s perform a single refresh() and retry.
-	return bt.inner.RoundTrip(in)
+	res, err := bt.inner.RoundTrip(in)
+
+	// Perform a token refresh() and retry the request in case the token has expired
+	if err == nil && res.StatusCode == http.StatusUnauthorized {
```

Comment:
I don't think we want to refresh and retry if `err == nil`, only if it was 401.

I'd also like to see some kind of reusable method we can call recursively to retry instead of setting up and calling `inner.RoundTrip` again.

Reply:
Makes sense, does an inline function that does prepares and sends the request sound reasonable to you?

something like this:
```
func (bt *bearerTransport) RoundTrip(in *http.Request) (*http.Response, error) {
  sendRequest := func() (*http.Response, error) {...}
  res, err := sendRequest()

  if err == nil && res.StatusCode == http.StatusUnauthorized {...}

  return res, err
}
```

Reply:
Yeah that's perfect. I might just short-circuit the error case:

```
res, err := sendRequest()
if err != nil {
  return nil, err
}
if res.StatusCode == http.StatusUnauthorized {
  bt.refresh()
  return sendRequest()
}
```